plugins/global-search/src/utils/indexer/types.ts

```diff
@@ -1,20 +1,28 @@
 import {
     type AnyNode,
+    ComponentInstanceNode,
     ComponentNode,
     isComponentInstanceNode,
     isComponentNode,
-    isFrameNode,
-    isSVGNode,
     isTextNode,
-    isWebPageNode,
+    TextNode,
     WebPageNode,
 } from "framer-plugin"
 
 export type IndexNodeRootNode = ComponentNode | WebPageNode
 export type IndexNodeRootNodeType = IndexNodeRootNode["__class"]
 
+export type IndexableNode = Extract<AnyNode, ComponentNode | TextNode | ComponentInstanceNode>
 export type IndexEntryType = AnyNode["__class"]
 
+export function isIndexableNode<T extends AnyNode>(value: T): value is T & IndexableNode {
+    if (isComponentInstanceNode(value)) return true
+    if (isTextNode(value)) return true
+    if (isComponentNode(value)) return true
```

Comment:
Is it on purpose that WebPageNode is no longer indexable here? If `ComponentNode` is indexable why not a webpage node here?

Reply:
I wonder if this should accept `AnyCanvasNode` to indicate root nodes should be excluded here?

Reply:
oh its just `CanvasNode` lol. but its no biggie just a nitpick

Reply:
leaving it at `AnyNode`, as otherwise, i would have to introduce an other type guard